### ISSUE TITLE
Handling 'trainable' issue

### ISSUE THREAD

**Issue by JaneyWit** [bstriner/keras-adversarial#22]
In the README file it is stated that the Adversarial Optimizer gets around the issue of setting network 'trainable' properties.  Can you explain how this works please? 
 In other sample code I have seen the discriminator model has to be compiled and then the 'trainable' property set to false before putting it into the GAN and compiling the GAN.  The discriminator weights are therefore not updated when GAN is being trained (this is my understanding). 
Can you please explain how it works in your implementation?  Thanks.

**Comment by bstriner** [comment]
TL;DR: `keras` models automatically get parameter lists depending on the `trainable` attribute at time of compilation but `keras-adversarial` models require you to provide the parameter lists to the constructor. If you want the same behavior as Keras you can use `trainable_weights` but there are much better ways to do it.

When you compile a normal Keras model it updates whatever parameters are currently set to trainable. That makes a stateful API, which tends to be finicky and harder to understand. That means to build a model with multiple players you have to set the state to the configuration for one player, compile, set the state for the next player, compile the next model, etc. That means the sequence of everything is important.

You end up with something like this.

```python
# do something to configure trainable settings for player 1
submodel1.trainable=False
submodel2.trainable=True
# make and compile model 1
model1 = Model(x, basemodel(x))
model1.compile(...)
# change state to configuration for player 2
submodel1.trainable=True
submodel2.trainable=False
# make and compile model 2
model2 = Model(x, basemodel(x))
model2.compile(...)
# some custom for loop is required
for i in range(epochs):
  model1.train_on_batch(...)
  model2.train_on_batch(...)
```

Internally, the `trainable_weights` attribute is dynamically determined based on the current setting of the `trainable` attribute.

`keras-adversarial` models require you to provide separate lists of parameters for each player.

If you wanted make a `keras-adversarial` model in the traditional stateful way, you could do something like this:

```python
# do something to configure trainable settings for player 1
submodel1.trainable=False
submodel2.trainable=True
# get player 1 weights
player_1 = basemodel.trainable_weights
# change state to configuration for player 2
submodel1.trainable=True
submodel2.trainable=False
# get player 2 weights
player_2 = basemodel.trainable_weights
# build adversarial model using players
adversarial_model = AdversarialModel(player_params=[player_1, player_2], ...)
adversarial_model.compile(...)
# combination of models handled by Adversarial Optimizer
# no custom iteration required
adversarial_model.fit(...)
```

However, the `AdversarialModel` doesn't require using the stateful method. For most adversarial models it is really easy to just make submodels or other ways to list the parameters.

This is equivalent to the above but a lot easier to understand. It also isn't stateful so the order of the lines doesn't matter so it should be harder to mess something up.

```python
player_1 = submodel2.trainable_weights
player_2 = submodel1.trainable_weights
adversarial_model = AdversarialModel(player_params=[player_1, player_2], ...)
```

For more complicated models you might concatenate the weights of several models to build the weights for each player. For example, one player is `player_1=encoder.trainable_weights+decoder.trainable_weights` and the other is `player_2=adversary.trainable_weights`.

Cheers
